src/SymmetricPositiveDefinite.jl

```diff
@@ -0,0 +1,176 @@
+using LinearAlgebra: svd, eig, eigen
+
+@doc doc"""
+    SymmetricPositiveDefinite{N} <: Manifold
+
+The manifold of symmetric positive definite matrices, i.e.
+
+```math
+\mathcal P(n) \coloneqq
+\bigl\{
+    x \in \mathbb R^{n\\times n} :
+    \xi^\mathrm{T}x\xi > 0 \text{ for all } \xi \in \mathbb R^{n}\backslash\{0\}
+\bigr\}
+```
+
+# Constructor
+
+    SymmetricPositiveDefinite(n)
+
+generates the $\mathcal P(n) \subset \mathbb R^{n\times n}$
+"""
+struct SymmetricPositiveDefinite{N} <: Manifold end
+SymmetricPositiveDefinite(n::Int) = SymmetricPositiveDefinite{n}()
+
+@doc doc"""
+    manifold_dimension(::SymmetricPositiveDefinite{N})
+
+returns the dimension of the manifold [`SymmetricPositiveDefinite`](@ref) $\mathcal P(n)$, N\in \mathbb N$, i.e.
+```math
+    \frac{n(n+1)}{2}    
+```
+"""
+@generated manifold_dimension(::SymmetricPositiveDefinite{N}) where {N} = N*(N+1)/2
+
+@doc doc"""
+
+"""
+struct LinearAffineMetric <: Metric end
+@traitimpl HasMetric{SymmetricPositiveDefinite, LogEuclidean}
```

Comment:
This means that `SymmetricPositiveDefinite` is shorthand for `MetricManifold{SymmetricPositiveDefinite, LogEuclidean}`, but `SymmetricPositiveDefinite` seems to be forwarding operations to `MetricManifold{SymmetricPositiveDefinite, LinearAffineMetric}`. Also, this line is repeated below, and it should be `LogEuclideanMetric`.

Reply:
Oh, then I misunderstood, how to define (a) a default and (b) the coupling of metric and manifold. I interpreted the trait as “Manifold A _has_ (or _may have_ to be precise) metric B”, your explanation says it's “The default metric for Manifold A is Metric B”, so then the default I am aiming for (since thats the classical one also giving us a Lie group) `HasMetric{SymmetricPositiveDefinite,LinearAffineMetric}`. I will change that. I was maybe confused, because some functions default also to a certain metric, for example in Euclidean I think.

Reply:
See my comment below. I think it actually makes sense for us to not use `HasMetric` very often and instead explicitly implement `MetricManifold` variants and forward to them if the overhead isn't high. It looks like it takes ~15ns to create a new `MetricManifold`.

Reply:
Ah and I had a copy paste error, the first traitsimpl should've coupled with `LinearAffineMetric`. I'll remove both in the next commit.